client/src/components/_common/Button/Button.test.js

```diff
@@ -1,20 +1,160 @@
+// WARNING: Relies on `Icon` because of `getByRole('img')`
 import React from 'react';
 import { render } from '@testing-library/react';
-import Button from './Button';
+import Button, * as BTN from './Button';
+
+import '@testing-library/jest-dom/extend-expect';
+
+const TEST_TEXT = '…';
+const TEST_TYPE = 'primary';
+const TEST_SIZE = 'medium';
+
+function testClassnamesByType(type, size, getByRole, getByTestId) {
+  const root = getByRole('button');
+  const text = getByTestId('text');
+  const typeClassName = BTN.TYPE_MAP[type];
+  const sizeClassName = BTN.SIZE_MAP[size];
+  expect(root.className).toMatch('root');
+  expect(root.className).toMatch(new RegExp(typeClassName));
+  expect(root.className).toMatch(new RegExp(sizeClassName));
+  expect(text.className).toMatch('text');
+}
+
+function muteTypeNotLinkNoSizeLog(type, size) {
+  if (type !== 'link' && !size) console.debug = jest.fn();
+}
+
+function isPropertyLimitation(type, size) {
+  let isLimited = false;
+
+  if (
+    type === 'primary' && size === 'small' ||
+    type !== 'link' && !size ||
+    type === 'link' && size
+  ) isLimited = true;
+
+  return isLimited;
+}
+
+function getExpectedType(type, size) {
+  let expType = type;
+  if (type === 'primary' && size === 'small') { expType = ''; }
+  return expType;
+}
 
 describe('Button', () => {
-  it('does not render button without text', () => {
-    const { queryByTestId } = render(
-      <Button data-testid="no button here"></Button>
+  it('uses given text', () => {
+    muteTypeNotLinkNoSizeLog();
+    const { getByTestId } = render(
+      <Button>{TEST_TEXT}</Button>
     );
-    const el = queryByTestId('no button here');
-    expect(el).toBeNull;
+    expect(getByTestId('text').textContent).toEqual(TEST_TEXT);
   });
-  it('disables button when in loading state', () => {
-    const { queryByText } = render(
-      <Button isLoading={true}>Loading Button</Button>
-    );
-    const el = queryByText('Loading Button');
-    expect(el).toBeDisabled;
```

Comment:
**Moved:** Loading tests are [near the bottom of the file](https://github.com/TACC/Core-Portal/pull/640/files#diff-9bad75f8a7092bc99d6151bfde064bf97e95ed116ccb3ed3d83b236911b90632R106-R123).